src/python/turicreate/toolkits/sound_classifier/_mx_sound_classifier.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+# Copyright © 2019 Apple Inc. All rights reserved.
+#
+# Use of this source code is governed by a BSD-3-clause license that can
+# be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+
+
+import mxnet as mx
+import time
+import turicreate as _tc
+
+from .._mxnet import _mxnet_utils
+
+class MultiLayerPerceptronMXNetModel():
+
+
+    def __init__(self, feature_output_length, num_labels, custom_layer_sizes, verbose):
+
+        self.ctx = _mxnet_utils.get_mxnet_context()
+        self.verbose = verbose
+        self.custom_NN = self._build_custom_neural_network(feature_output_length, num_labels, custom_layer_sizes)
+        self.custom_NN.initialize(mx.init.Xavier(), ctx=self.ctx)
+
+        self.trainer = mx.gluon.Trainer(self.custom_NN.collect_params(), 'nag', {'learning_rate': 0.01, 'momentum': 0.9})
+        self.softmax_cross_entropy_loss = mx.gluon.loss.SoftmaxCrossEntropyLoss()
+
+
+    def train(self, data, label):
+        """
+        Parameters
+        ----------
+        data : NumPy Array
+
+        label : NumPy Array
+        """
+
+        # Inside training scope
+        data_shape = data.shape[0]
+        data = mx.gluon.utils.split_and_load(data, ctx_list=self.ctx, batch_axis=0, even_split=False)
+        label = mx.gluon.utils.split_and_load(label, ctx_list=self.ctx, batch_axis=0, even_split=False)
+        with mx.autograd.record():
+            for x, y in zip(data, label):
+                z = self.custom_NN(x)
+                # Computes softmax cross entropy loss.
+                loss = self.softmax_cross_entropy_loss(z, y)
+                # Backpropagate the error for one iteration.
+                loss.backward()
+        # Make one step of parameter update. Trainer needs to know the
+        # batch size of data to normalize the gradient by 1/batch_size.
+        self.trainer.step(data_shape)
```

Comment:
This comment describes the information being passed to `self.trainer.step` as the batch size. Why not just call the variable `batch_size`, since by construction we know it is a scalar value, the first element of the data's shape?

Reply:
Sure I can make it `batch_size = min(batch_size, data.shape[0])` since `data.shape[0]` can be smaller than batch_size if examples it holds are fewer.